src/inmanta/server/services/metricservice.py

```diff
@@ -59,3 +64,70 @@ def start_metric_reporters(self) -> None:
                 tags=opt.influxdb_tags.get(),
             )
             self._influx_db_reporter.start()
+
+    def start_auto_benchmark(self):
+        gauge("self.spec.cpu", CPUMicroBenchMark())
+
+
+class CachingCallbackGuage(Gauge):
+    """
+    A gauge that calculates a value on the fly and holds it for a specific time interval.
+
+    Intended to keep load under control.
+
+    Be aware that the callback has to be short, as it is called from the IOLoop main thread.
+    """
+
+    def __init__(self, interval: int = 1):
+        self.next_time: float = 0
```

Comment:
Should we call `__init__()` in the superclass.

Reply:
done